lib/ansible/playbook/role/definition.py

```diff
@@ -54,6 +56,7 @@ def __init__(self, play=None, role_basedir=None, variable_manager=None, loader=N
         self._role_path = None
         self._role_basedir = role_basedir
         self._role_params = dict()
+        self._collection_list = collection_list
```

Comment:
```suggestion
        self._collection_list = [] if collection_list is None else collection_list
```

Reply:
We may actually support a difference between `None` (use the old behavior) and "empty list" (require FQ everything) in the future, so I'd rather not always convert those internally to `[]`. `None` is by far the most common tested case (basically everything except the `collections` test target), so the code's already well-exercised checking via `not` before we go enumerating it.